v-next/hardhat/src/internal/builtin-plugins/solidity/build-system/compiler/solcjs-wrapper.ts

```diff
@@ -0,0 +1,138 @@
+// This wrapper was created by extracting the parts of the solc-js package
+// (https://github.com/ethereum/solc-js) that we need to perform compilation.
+
+import * as semver from "semver";
+
+interface Solc {
+  cwrap<T>(ident: string, returnType: string | null, argTypes: string[]): T;
+
+  // eslint-disable-next-line @typescript-eslint/naming-convention -- this is a C function
+  _solidity_reset?: Reset | null;
+  // eslint-disable-next-line @typescript-eslint/naming-convention -- this is a C function
+  _solidity_version?: Version | null;
+  // eslint-disable-next-line @typescript-eslint/naming-convention -- this is a C function
+  _version?: Version | null;
+  // eslint-disable-next-line @typescript-eslint/naming-convention -- this is a C function
+  _compileStandard?: Compile | null;
+  // eslint-disable-next-line @typescript-eslint/naming-convention -- this is a C function
+  _solidity_compile?: Compile | null;
+}
+
+type Reset = () => string;
+type Version = () => string;
+type Compile = (
+  input: string,
+  callbackPtr: number | null,
+  callbackContextPtr?: null,
```

Comment:
This is an odd type, but we either want to skip this argument or provide an explicit null. That depends on the version of solc that we're going to be using.